### PR TITLE
Add --configmap to podman-remote kube play

### PR DESCRIPTION
**Pull request by umohnani8**
Enable the --configmap flag for the remote case of podman
kube play. Users can pass in the paths to the configmap files
for kube play to use when creating the pods and containers from
a kube yaml file. The configmap file is read and the contents are
appended to the contents of the main yaml file before passed to the
remote client.

Fixes https://github.com/containers/podman/issues/17513

#### Does this PR introduce a user-facing change?

```release-note
Add --configmap flag to podman-remote kube play.
```

### CODE REVIEW

**Comment by umohnani8** [comment]
@vrothberg @baude was there a reason that this flag was disabled for the remote case? I tested it out locally and it works fine with podman-remote but maybe there is something I am missing.

**Comment by vrothberg** [comment]
> @vrothberg @baude was there a reason that this flag was disabled for the remote case? I tested it out locally and it works fine with podman-remote but maybe there is something I am missing.

It probably only works because you test it on the same machine.  Passing a _path_ to the server will work in this scenario but what if the client is on another machine? What would need to happen is to tar up the file, send it over to the server and use it there.

There are other flags/features (e.g., build) that don't (yet?) work for the same reason.

**Comment by Luap99** [comment]
Yes this came up before and we should not send local paths via API.
Play kube already supports configmaps as part of the same (mutli doc) yaml file, I think we should patch the remote client to read the configmaps and append them to the original yaml file which is then send via the body to the server.

For historical reference: https://github.com/containers/podman/pull/12243

File: pkg/api/server/register_kube.go
Comment (Luap99) [inline pkg/api/server/register_kube.go:132]: This is not API parameter so remove it.

**Review by Luap99 — COMMENTED** [review]
There are more SkipIfRemote's that should be removed in the e2e tests.

File: test/e2e/play_kube_test.go
Comment (umohnani8) [reply test/e2e/play_kube_test.go:4411]: fixed

**Review by Luap99 — COMMENTED** [review]
LGTM, @vrothberg PTAL

**Review by vrothberg — APPROVED** [review]
/lgtm
/hold

File: cmd/podman/kube/play.go
Comment (TomSweeneyRedHat) [inline cmd/podman/kube/play.go:165]: Nit, but it probably ought to have Kubernetes
```suggestion
	flags.StringSliceVar(&playOptions.ConfigMaps, configmapFlagName, []string{}, "`Pathname` of a YAML file containing a Kubernetes configmap")
```

**Comment by umohnani8** [comment]
@vrothberg  @Luap99 Can I please get a lgtm here

File: cmd/podman/kube/play.go
Comment (umohnani8) [reply cmd/podman/kube/play.go:165]: Will fix in a follow up PR.

**Comment by umohnani8** [comment]
@containers/podman-maintainers can I please get a lgtm here

**Pull request merged**
